realsense2_camera/include/base_realsense_node.h

```diff
@@ -238,6 +260,7 @@ namespace realsense2_camera
         void updateSensors();
         void publishServices();
         void startPublishers(const std::vector<rs2::stream_profile>& profiles, const RosSensor& sensor);
+        void startRGBDPublisherIfNeeded();
```

Comment:
why is this publisher different and has its own function?

Reply:
because each publisher was related to sensor.
This publisher is unique and depends on two sensors (and on their ON/OFF status)